### PR TITLE
fix: use deviceId instead of uid for uninstallation event mapping

### PR DESCRIPTION
**Pull request by nsrCodes**
The locally stored user id has many problems:
1. not being cleared when logged out
2. Randomly getting generated when not found (which is just generating fake new user data)
3. Mapping events from an anon user to a random uid 

shifting to using the deviceId instead

### CODE REVIEW

File: app/public/index.html [hunk 1]
```diff
@@ -11,7 +11,8 @@
         window.location.href = 'https://app.requestly.io';
       }
 
-      if ("https://app.requestly.io/goodbye/" === window.location.href) { const a = { api_key: "62ff1b46909e50358cfca0668d41f011", events: [{ user_id: localStorage.getItem("__rq_uid") || Math.random().toString(36).slice(2, 10), event_type: "extension_uninstalled" }] }, b = new Blob([JSON.stringify(a)]); navigator.sendBeacon("https://api2.amplitude.com/2/httpapi", b) }
+      const deviceId = localStorage.getItem("__rq_device_id")
+      if ("https://app.requestly.io/goodbye/" === window.location.href && deviceId) { const a = { api_key: "62ff1b46909e50358cfca0668d41f011", events: [{ device_id: deviceId, event_type: "extension_uninstalled" }] }, b = new Blob([JSON.stringify(a)]); navigator.sendBeacon("https://api2.amplitude.com/2/httpapi", b) }
```
Comment (wrongsahil) [inline app/public/index.html:15]: We should use uid first and if it is not found, then we should use the deviceId.

Comment (wrongsahil) [inline app/public/index.html:15]: We need to set `extension_uninstall_date` attribute too on uninstall

File: app/public/index.html [hunk 2]
```diff
@@ -11,7 +11,8 @@
         window.location.href = 'https://app.requestly.io';
```
Comment (wrongsahil) [inline app/public/index.html:11]: This file is deprecated when we shifter to vite. Please delete this file to avoid confusion in future

**Review by wrongsahil — CHANGES_REQUESTED** [review]

File: app/public/index.html (hunk 1 above)
Comment (nsrCodes) [reply app/public/index.html]: setting attributes is not available in the http api

**Review by wrongsahil — APPROVED** [review]

**Pull request merged**